test/e2e/categories_test.go

```diff
@@ -0,0 +1,186 @@
+		clusterName      string
+		clusterResources *clusterctl.ApplyClusterTemplateAndWaitResult
+		cancelWatches    context.CancelFunc
+	)
+	BeforeEach(func() {
+		clusterName = generateTestClusterName(specName)
+		clusterResources = new(clusterctl.ApplyClusterTemplateAndWaitResult)
+		Expect(bootstrapClusterProxy).NotTo(BeNil(), "BootstrapClusterProxy can't be nil")
+		namespace, cancelWatches = setupSpecNamespace(ctx, specName, bootstrapClusterProxy, artifactFolder)
+	})
+	AfterEach(func() {
+		dumpSpecResourcesAndCleanup(ctx, specName, bootstrapClusterProxy, artifactFolder, namespace, cancelWatches, clusterResources.Cluster, e2eConfig.GetIntervals, skipCleanup)
+	})
+
+	It("Create a cluster with default cluster categories (no additional categories)", func() {
+		Expect(namespace).NotTo(BeNil())
+		flavor := clusterctl.DefaultFlavor
+		By("Creating a workload cluster (no additional categories)")
+		deployClusterAndWait(
+			deployClusterParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				flavor:                flavor,
+				clusterctlConfigPath:  clusterctlConfigPath,
+				artifactFolder:        artifactFolder,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+				e2eConfig:             *e2eConfig,
+			}, clusterResources)
+
+		By("Checking cluster category condition is true")
+		verifyConditionOnNutanixCluster(verifyConditionOnNutanixClusterParams{
+			clusterName:           clusterName,
+			namespace:             namespace,
+			bootstrapClusterProxy: bootstrapClusterProxy,
+			expectedCondition: clusterv1.Condition{
+				Type:   infrav1.ClusterCategoryCreatedCondition,
+				Status: corev1.ConditionTrue,
+			},
+		})
+
+		By("Checking if a category was created")
+		expectedCategoryKey := getExpectedClusterCategoryKey(clusterName)
+		_, err := nutanixClient.V3.GetCategoryValue(expectedCategoryKey, defaultClusterCategoryValue)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("Checking if there are VMs assigned to this category")
+		expectedCategories := map[string]string{
+			expectedCategoryKey: defaultClusterCategoryValue,
+		}
+		verifyCategoriesNutanixMachines(clusterName, namespace.Name, expectedCategories)
+
+		By("PASSED!")
+	})
+	It("Create a cluster with additional categories", func() {
+		Expect(namespace).NotTo(BeNil())
+		flavor := "additional-categories"
+
+		By("Creating a workload cluster with additional categories set")
+		deployClusterAndWait(
+			deployClusterParams{
+				clusterName:           clusterName,
+				namespace:             namespace,
+				flavor:                flavor,
+				clusterctlConfigPath:  clusterctlConfigPath,
+				artifactFolder:        artifactFolder,
+				bootstrapClusterProxy: bootstrapClusterProxy,
+				e2eConfig:             *e2eConfig,
+			}, clusterResources)
+
+		By("Assign additional categories to the vms")
+		expectedCategoryKey := getExpectedClusterCategoryKey(clusterName)
+		additionalCategory := getAdditionalCategoryFromEnvironment()
```

Comment:
Since this is a e2e test code, can we just hardcode the additional category instead of reading it from the env?